### PR TITLE
leaf refresh enhancement

### PR DESCRIPTION
**Pull request by qpdpQ**
Signed-off-by: qpdpQ <liyuchen223@gmail.com>

### CODE REVIEW

**Comment by qpdpQ** [comment]
How to test it:
1. use this image `quay.io/yuchen_li1/ibm-cert-manager-operator-amd64@sha256:1e2330b25f63e1e5810ff0f89f4a52d7342a4a39350f8455b403c7bc55ecfd25` in ibm-cert-manager-operator
2. create a test certificate chain(you can use the file in the following comment)
3. change some data in the `test-ca-certificate` and watch if there is any change in `test-leaf-certificate`
4. delete `test-ca-certificate` and change some data in the `test-ca-certificate-secret` and watch if there is any change in `test-leaf-certificate`

File: controllers/cert-manager/certificaterefresh_controller.go [hunk 1]
```diff
@@ -76,9 +75,9 @@ func (r *CertificateRefreshReconciler) Reconcile(ctx context.Context, req ctrl.R
 	reqLogger.Info("Reconciling CertificateRefresh")
 
 	// Get the certificate that invoked reconciliation is a CA in the listOfCAs
-
-	cert := &certmanagerv1.Certificate{}
-	err := r.Client.Get(context.TODO(), req.NamespacedName, cert)
+	secret := &corev1.Secret{}
+	//cert := &certmanagerv1.Certificate{}
```
Comment (bitscuit) [inline controllers/cert-manager/certificaterefresh_controller.go:79]: remove unused code

File: controllers/cert-manager/certificaterefresh_controller.go [hunk 2]
```diff
@@ -109,6 +108,18 @@ func (r *CertificateRefreshReconciler) Reconcile(ctx context.Context, req ctrl.R
...
+	// Get the certificate by this secret in the same namespace
+	cert, err := r.getCertificateBySecret(secret)
+	foundCert := true
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logd.Info("Cert instance not found in the namespace")
+			foundCert = false
+		} else {
+			return ctrl.Result{}, err
+		}
```
Comment (bitscuit) [inline controllers/cert-manager/certificaterefresh_controller.go:120]:
```suggestion
		if !errors.IsNotFound(err) {
			return ctrl.Result{}, err
		}
		logd.Info("Failed to find backing Certificate object for secret", "name:", secret.Name, "namespace:", secret.Namespace)
		foundCert = false
```

File: controllers/cert-manager/certificaterefresh_controller.go [hunk 3]
```diff
@@ -156,40 +167,43 @@ func (r *CertificateRefreshReconciler) Reconcile(ctx context.Context, req ctrl.R
...
-	found := false
-	for _, caCert := range listOfCAs {
-		if caCert.CertName == cert.Name && caCert.Namespace == cert.Namespace {
-			found = true
-			break
+	// found ca cert or ca secret
+	foundCA := false
+	// if we found this certificate in the same namespace
+	if foundCert {
+		// check this cert is ca or not
```
Comment (bitscuit) [inline controllers/cert-manager/certificaterefresh_controller.go:174]:
```suggestion
		// check if certificate is in list of CAs to refresh
```

Comment (bitscuit) [reply controllers/cert-manager/certificaterefresh_controller.go:174]: comment was misleading before

File: controllers/cert-manager/certificaterefresh_controller.go (hunk 2 above)
Comment (bitscuit) [reply controllers/cert-manager/certificaterefresh_controller.go:120]: simplify conditional logic

**Comment by qpdpQ** [comment]
changed the order of label check

**Comment by qpdpQ** [comment]
this is the new image`quay.io/yuchen_li1/ibm-cert-manager-operator-amd64@sha256:df8bdddc8385e42c26d32be8dbb4c4a7765ec25ed1b07bf2f189151fa7692ee3`

**Review by bitscuit — APPROVED** [review]
/lgtm

**Pull request merged**